src/darwin/Framework/CHIP/CHIPOnboardingPayloadParser.m

```diff
@@ -0,0 +1,60 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "CHIPOnboardingPayloadParser.h"
+#import "CHIPManualSetupPayloadParser.h"
+#import "CHIPQRCodeSetupPayloadParser.h"
+#import "CHIPSetupPayload.h"
+
+@implementation CHIPOnboardingPayloadParser
+
++ (CHIPSetupPayload * __nullable)setupPayloadForOnboardingPayload:(NSString *)onboardingPayload
+                                                           ofType:(CHIPOnboardingPayloadType)type
+                                                            error:(NSError * __autoreleasing *)error
+{
+    CHIPSetupPayload * payload;
+    switch (type) {
+    case CHIPOnboardingPayloadTypeManualCode:
+        payload = [self setupPayloadForManualCodeOnboardingPayload:onboardingPayload error:error];
+        break;
+    case CHIPOnboardingPayloadTypeQRCode:
+    case CHIPOnboardingPayloadTypeTomatoEnhanced:
```

Comment:
I believe the payload returned by the TomatoEnhanced API is a 11-digit short code, so we should move this next to the ManualCode case statement.

Reply:
yes verified and it is generated with the manualcodeparserm changed it